File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem56_KokoEatingBananas.java
package org.phoenix.leetcode.challenges;

public class Problem56_KokoEatingBananas {
    public int minEatingSpeed(int[] piles, int h) {
        int left = 0;
        int right = 1000000000;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            int hour = 0;
            for (int pile : piles) {
                hour += Math.ceil((double) pile / mid);
            }
            if (hour <= h) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return left;
    }
}

File: JavaScript/leetcode75/src/main/Problem56_KokoEatingBananas.js
/**
 * @param {number[]} piles
 * @param {number} h
 * @return {number}
 */
var minEatingSpeed = function (piles, h) {
  let left = 0;
  let right = 1000000000;
  while (left <= right) {
    let mid = Math.floor(left + (right - left) / 2);
    let hour = 0;
    for (let pile of piles) {
      let div = Math.ceil(pile / mid);
      hour += div;
    }
    if (hour <= h) {
      right = mid - 1;
    } else {
      left = mid + 1;
    }
  }
  return left;
};

module.exports = minEatingSpeed;
